server/testutils/containere2e/apimock.go

```diff
@@ -126,6 +145,57 @@ func (m *MockClient) Delete(id string, url string, body map[string]any) error {
 	return m.Mock(http.MethodDelete, id, url, http.StatusOK, body)
 }
 
+func (m *MockClient) MockBatch(id string, requests []BatchRequest, responses []BatchResponse) error {
+	for idx := range responses {
+		responses[idx].ID = "{{#jsonPath}}$.requests[" + fmt.Sprint(idx) + "].id{{/jsonPath}}{{jsonPathResult}}"
+	}
+
+	responseData, err := json.Marshal(map[string]any{
+		"statusCode": 200,
+		"body": map[string]any{
+			"type":        "JSON",
+			"contentType": "application/json",
+			"string":      map[string]any{"responses": responses},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	mockExpectation := map[string]any{
+		"id":       id,
+		"priority": 10,
+		"httpRequest": map[string]any{
+			"method": http.MethodPost,
+			"path":   "/v1.0/$batch",
+			"body": map[string]any{
+				"type":      "JSON",
+				"matchType": "ONLY_MATCHING_FIELDS",
+				"json":      map[string][]BatchRequest{"requests": requests},
+			},
+		},
+		"httpResponseTemplate": map[string]any{
+			"templateType": "MUSTACHE",
+			"template":     string(responseData),
+		},
+	}
+
+	testMock, err := json.Marshal(mockExpectation)
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequest("PUT", m.api+"/mockserver/expectation", bytes.NewReader(testMock))
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
```

Comment:
Any reason to close the body here? This is just for my own understanding as I'm doing something similar with the load test

Reply:
That is expected in go to avoid some leaks that can happen. Whenever you do a request, you should close the body of the response, it is recommended even if you don't read it.